### DISCUSSION TITLE
How do I deserialize a serialized machine?

### DISCUSSION THREAD

**Discussion by dabrady** [statelyai/xstate#4785] [Q&A] [answered]
> Disclaimer: I've only been exploring XState for about an hour or two, but I can't find the answer to my question on the docs or any existing Issue/Discussion.

**XState Version:** 5.9.1

Aloha folks,

How do I deserialize a machine from JSON back into a usable machine? My first attempt (simply JSON parsing the JSONified output of `StateMachine#toJSON()` and feeding it back into `createMachine`) works up until you try to use the deserialized machine in an actor and start it up, at which point it dies horribly:

```javascript
$ node
Welcome to Node.js v20.9.0.
Type ".help" for more information.
> const { createActor, setup } = require('xstate');
> var config = {
  id: 'le machine',
  initial: 'Start',
  states: { Start: { on: { end: 'End' } }, End: { on: { start: 'Start' } } },
};
> createActor(setup({}).createMachine(config)).start(); // works as expected
// undefined
> var serializedMachine = JSON.stringify(createMachine(config).toJSON());
> var deserializedMachine = JSON.parse(serializedMachine);
> createActor(setup({}).createMachine(deserializedMachine)).start(); // dies horribly
// Uncaught:
// Error: Initial state node "[object Object]" not found on parent state node #le machine
// ...<stack trace>
```

I don't see any dedicated API methods catering to restoring serialized machines, so I presume I'm either missing something obvious, or it's not possible.

I don't have a concrete use-case for this yet by the way, just vague ideas; I simply thought it ought to be possible, otherwise the only obvious use-case for the `.toJSON` method is logging.

**Comment by davidkpiano** [comment, marked as answer]
Hey, you're right; it is possible. Apologies for the lack of documentation on this.

You should stringify `machine.config` instead of `machine.toJSON()`.

**Reply by dabrady** [reply]
No worries, thank you for the quick response! 🙏🏻

**Reply by dabrady** [reply]
While I have you here, can you explain the intended use-case(s) of the `StateMachine#toJSON` function, if it's not meant to persist the state machine?
